### PR TITLE
Don't open tree-view to auto-reveal files

### PR DESCRIPTION
**Pull request by BinaryMuse**
This PR changes tree-view such that the dock is not expanded automatically when files are revealed due to the Auto Reveal option being checked.

The matrix of behavior for various states of **Auto Reveal** and **Focus On Reveal** settings is now:

|            Auto Reveal / Focus On Reveal ➡️  | **off**/**off** | **on**/**off** | **off**/**on** | **on**/**on** |
|-----------------------------------------|-----------------|----------------|----------------|---------------|
| *when changing pane items*               |                 |                |                |               |
| tree-view opens if closed                | ❌               | :x:             | :x:            | :x:            |
| active file is revealed                  | :x:              | :white_check_mark:             | :x:             | :white_check_mark:           |
| tree-view is focused                     | :x:              | :x:             | :x:             | :x:            |
| *when running Reval Active File command* |                 |                |                |               |
| tree-view opens if closed                | :white_check_mark:             | :white_check_mark:            | :white_check_mark:            | :white_check_mark:           |
| active file is revealed                  | :white_check_mark:             | :white_check_mark:            | :white_check_mark:            | :white_check_mark:           |
| tree-view is focused                     | :x:              | :x:             | :white_check_mark:            | :white_check_mark:           |

In short:

1. The tree-view will **only** be shown when running the Reveal Active File command
2. The tree-view will **only** be focused when the Reval Active File command is run **and** Focus On Reveal is on

Fixes https://github.com/atom/tree-view/issues/796
Closes https://github.com/atom/tree-view/pull/983

### CODE REVIEW

**Comment by BinaryMuse** [comment]
Tests on Windows are crashing due to https://github.com/atom/tree-view/issues/1203 which is resolved by https://github.com/bruce/node-temp/pull/69 (and a dependency bump).

**Comment by BinaryMuse** [comment]
These tests crash on Windows on `master` and pass on Travis, so I'm gonna go ahead and merge this.
